rfcs/2021-09-01-8547-accept-metrics-in-datadog-agent-source.md

```diff
@@ -0,0 +1,168 @@
+     Datadog Agent would need it to migrate to Vector aggegation.
+
+**Regarding the tagging issue:** A -possibly temporary- work-around would be to store incoming tags with the complete
+"key:value" string as the key and an empty value to store those in the extisting map Vector uses to store
+[tags](https://github.com/timberio/vector/blob/master/lib/vector-core/src/event/metric.rs#L60) and slightly rework the
+`datadog_metrics` sink not to append `:` if a tag key has the empty string as the corresponding value.
+
+
+## Rationale
+
+- Smoother Vector integration with Datadog.
+- Needed for Vector to act as a complete Datadog Agent aggregator (but further work will still be required).
+- Extend the Vector ecosystem, bring additional feature for distribution metrics that would enable consistent
+  aggregation.
+
+## Drawbacks
+
+- It arbitrary imposes internal sketches representation to follow the DDSketch paper.
+
+## Prior Art
+
+- Existing Vector metrics source over HTTP(s) works well, those constructs will be reused
+- Datadog logs is already supported in the `datadog_agent` source
+- Vector has its own vector-to-vector protocol that serve a similar purpose but the idea is to enable Vector to receive
+  Datadog metrics from unmodified Datadog Agents so that is probably not an option
+- There are many [ways](https://edoliberty.github.io/papers/streamingQuantiles.pdf) of implementing quantile sketches
+
+## Alternatives
+
+- Use an alternate protocol between Datadog Agents and Vector (Like Prometheus, Statds or Vector own protocol)
+- For sketches, we could flatten sketches and compute usual derived metrics (min/max/average/count/some percentile) and
+  send those as gauge/count, but it would prevent (or at least impact) existing distribution/sketches user
+
+## Outstanding Questions
+
+- Datadog Agent configuration, diverting metrics to a custom endpoints will likely divert other kind of traffic, this
+  need to be clearly identified / discussed
+- Is there any other metrics type that could required some degree of adaptation, there is the non monotonic counter that
+  exists in the agent but not in Vector that may require the introduction of a so-called `NonMonotonicCounter` in
+  Vector.
+
+## Plan Of Attack
+
+- [ ] Implement the proxy logic in the `datadog_agent` source
+- [ ] Support `/api/v1/series` route still in the `datadog_agent` source, implement complete support in the
+  `datadog_metrics` sinks for the undocumented fields, incoming tags would be stored as key only with an empty string
+  for their value inside Vector. Validate the `Agent->Vector->Datadog` scenario for gauge, count & rate.
+- [ ] Extend Vector to support sketches following the DDSketch paper, implement sketches forwarding in the
+  `datadog_metrics` sinks.
+- [ ] Support `/api/beta/sketches` route, again in the `datadog_agent`, and validate the `Agent->Vector->Datadog`
```

Comment:
This is out-of-scope for this RFC that is source focused, but I just wanted to call out that we'll need to also augment `datadog_metrics` to support sending sketches.